backend/database_handler/accounts_manager.py

```diff
@@ -1,87 +1,89 @@
+        )
+        self.session.add(account_state)
+        self.session.commit()
+
     def is_valid_address(self, address: str) -> bool:
         return is_valid_address(address)
 
-    def get_account(self, account_address: str):
-        """Private method to retrieve if an account from the data base"""
-        condition = f"id = '{account_address}'"
-        account_data = self.db_client.get(self.db_accounts_table, condition)
-        return self._parse_account_data(account_data[0]) if account_data else None
+    def _get_account(self, account_address: str) -> CurrentState | None:
+        """Private method to retrieve an account from the data base"""
+        account = (
+            self.session.query(CurrentState)
+            .filter(CurrentState.id == account_address)
+            .one_or_none()
+        )
+        return account
 
-    def get_account_or_fail(self, account_address: str):
+    def get_account_or_fail(self, account_address: str) -> dict:
         """Private method to check if an account exists, and raise an error if not."""
-        account_data = self.get_account(account_address)
+        account_data = self._get_account(account_address)
         if not account_data:
             raise AccountNotFoundError(
                 account_address, f"Account {account_address} does not exist."
             )
-        return account_data
-
-    def register_new_account(self, address: str, balance: int) -> None:
-        account_state = {
-            "id": address,
-            "data": json.dumps({"balance": balance}),
-        }
-        self.db_client.insert(self.db_accounts_table, account_state)
+        return self._parse_account_data(account_data)
 
     def fund_account(self, account_address: str, amount: int):
         # account creation or balance update
-        account_data = self.get_account(account_address)
-        if account_data:
+        account_data = self._get_account(account_address)
+        if account_data is not None:
             # Account exists, update it
-            update_condition = f"id = {account_address}"
-            new_balance = account_data["data"]["balance"] + amount
-            updated_account_state = {
-                "data": json.dumps({"balance": new_balance}),
-            }
-
-            self.db_client.update(
-                self.db_accounts_table, updated_account_state, update_condition
-            )
+            # Dicts are mutable objects, we need to do something in order for SQLAlchemy to track their changes https://docs.sqlalchemy.org/en/20/orm/extensions/mutable.html
+            # In this case we are copying the dict, updating it and assigning it back to the object, which will trigger the change tracking
+            # I (Agustín Díaz) like this approach better than using `MutableDict` because it's more explicit than using the MutableDict class, and also doesn't leak the object to the rest of the code (which can be dangerous given that it's mutable)
```

Comment:
@AgustinRamiroDiaz I'm not sure if I follow, let's discuss this

Reply:
Sure!
For anyone else reading: sqlalchemy tracks changes in your models, so that when you `session.commit()` it knows which SQL queries it needs to run
The thing is that with mutable objects (like dictionaries) SQLAlchemy is not able to track its changes, given that SQLAlchemy only knows about the reference
Some solutions possible to this are:
a) copy the mutable object and reassign it (what is done in the code): this creates a new reference so SQLAlchemy can track it
b) use SQLAlchemy's `MutableDict`: this is a custom struct from SQLAlchemy which its able to be tracked by changes in its fields. The problem with using this is that if you expose this dictionary to another function/module and they mutate the dictionary, those changes will get reflected in the database. This is dangerous IMO because one could forget about this, expose the `CurrentState` object and other modules could modify its fields, not knowing they are actually modifying the DB